modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/GridCachePreloadLifecycleAbstractTest.java

```diff
@@ -160,14 +160,11 @@ public MyStringKey(String str) {
 
         /** {@inheritDoc} */
         @Override public boolean equals(Object obj) {
-//            if (this == obj)
-//                return true;
-//
-//            if (obj == null)
-//                return false;
-//
-//            if (getClass() != obj.getClass())
-//                return false;
+            if (this == obj)
```

Comment:
I think it's wrong.
Maybe state when a.hashCode() != b.hashCode() but a.equals(b) is true.